name-service/program/src/instruction.rs

```diff
@@ -89,6 +89,20 @@ pub enum NameRegistryInstruction {
     ///   2. `[writeable]` Refund account
     ///
     Delete,
+
+    /// Realloc the data of a name record.
+    ///
+    /// The space change cannot be more than `MAX_PERMITTED_DATA_LENGTH` different than
+    /// current `space`.
+    ///
+    /// Accounts expected by this instruction:
+    ///   0. `[writeable]` Name record to be reallocated
+    ///   1. `[signer]` Account owner
+    ///
+    Realloc {
+        /// Number of bytes of memory to allocate in addition to the `NameRecordHeader`.
```

Comment:
nit so we're sure it's the new size, and not the additional amount
```suggestion
        /// New total number of bytes in addition to the `NameRecordHeader`.
```

Also, let's add a warning that there are no checks on the data currently in the name record, so data may be truncated